Core.DomainModel/Organization/Organization.cs

```diff
@@ -295,29 +273,55 @@ public Maybe<UIModuleCustomization> GetUiModuleCustomization(string module)
         {
             if (root == null) throw new ArgumentNullException(nameof(root));
 
-            IExternalOrganizationalHierarchyUpdateStrategy strategy;
-            //Pre-validate
+            return GetStsOrganizationConnection(origin)
+                .Bind
+                (
+                    connection =>
+                    {
+                        var strategy = connection.GetUpdateStrategy();
+
+                        var childLevelsToInclude =
+                            levelsIncluded.Select(levels => levels - 1); //subtract the root level before copying
+                        var filteredTree = root.Copy(childLevelsToInclude);
+                        StsOrganizationConnection.SynchronizationDepth =
+                            levelsIncluded.Match(levels => (int?) levels, () => default);
+                        StsOrganizationConnection.SubscribeToUpdates = subscribeToUpdates;
+
+                        return strategy.PerformUpdate(filteredTree);
+                    }
+                );
+        }
+
+        public Result<StsOrganizationConnectionAddNewLogsResult, OperationError> AddExternalImportLog(OrganizationUnitOrigin origin,
+            StsOrganizationChangeLog changeLogToAdd)
+        {
+            return GetStsOrganizationConnection(origin)
+                .Bind<StsOrganizationConnectionAddNewLogsResult>(connection => connection.AddNewLogs(changeLogToAdd.WrapAsEnumerable()));
```

Comment:
why is the log as itself added as an enumereble?,,, a log entry is a single thing right?

It then has a collection of consequences inside